src/WebJobs.Script/Workers/MessageExtensions/ScriptInvocationContextExtensions.cs

```diff
@@ -34,9 +34,14 @@ public static async Task<InvocationRequest> ToRpcInvocationRequest(this ScriptIn
 
             foreach (var input in context.Inputs)
             {
-                if (!rpcValueCache.TryGetValue(input.val, out TypedData rpcValue))
+                TypedData rpcValue = null;
+                if (input.val == null || !rpcValueCache.TryGetValue(input.val, out rpcValue))
```

Comment:
shouldn't this be `input.val != null` ? otherwise key for trygetvalue could be null

Reply:
Due to short circuiting, the TryGetValue will only be called when value is NOT null. If null, we skip that check and just produce the rpc value.

Reply:
`rpcValueCache` is a dictionary and tryGetValue will throw argument exception if key is null right? Adding the check for !=null will avoid calling ToRpc altogether.

Reply:
The way this code is structured, we will never call TryGetValue with a null value due to the first check. If it's null, we short circuit. My tests prove this and exercise it fully. I can't have a != null check, because we still want to execute the if block for null values, to produce the default/empty rpc value.